environments/flow/recommended.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Js-coding-standards
+ *
+ * @author      Robert Rossmann <rr.rossmann@me.com>
+ * @copyright   2016 STRV
+ * @license     http://choosealicense.com/licenses/bsd-3-clause  BSD-3-Clause License
+ */
+
+'use strict'
+
+module.exports = {
+  plugins: ['flowtype'],
+
+  rules: {
+    // Marks Flow type identifiers as defined.
+    // Used to suppress no-undef reporting of type identifiers.
+    'flowtype/define-flow-type': 'warn',
+
+    // Enforces consistent use of trailing commas in Object and Tuple annotations.
+    // This rule takes one argument which mirrors ESLint's default comma-dangle rule.
+    'flowtype/delimiter-dangle': ['warn', 'always-multiline'],
+
+    // Enforces consistent spacing within generic type annotation parameters.
+    'flowtype/generic-spacing': ['error', 'always'],
+
+    // Checks for duplicate properties in Object annotations.
+    'flowtype/no-dupe-keys': 'error',
+
+    // Disallows Flow type imports, exports, aliases, and annotations in files missing a valid
+    // Flow file declaration (or a @noflow annotation).
+    'flowtype/no-types-missing-file-annotation': 'error',
+
+    // Enforces consistent separators between properties in Flow object types.
+    'flowtype/object-type-delimiter': ['error', 'comma'],
+
+    // Requires that all function parameters have type annotations.
+    'flowtype/require-parameter-type': [
+      'error',
+      {
+        excludeArrowFunctions: 'expressionsOnly',
```

Comment:
Weeell, I am not sure this is a good idea... Thinking about various array functions, like `.reduce()`, `.filter()`, callbacks etc.

Let's just remove this exception for now and we willl re-visit later if this turns out to be too much pain.

Reply:
This is one of most important rules here, we can set it to ignore arrow function (more [here](https://github.com/gajus/eslint-plugin-flowtype#require-parameter-type)), but I would not remove it at all.

Then this function is valid:
```javascript
function composeParams(params: { [string]: string }): string {
  return `?${Object.keys(params)
    .map(
      key => params[key] ? `${encodeURIComponent(key)}=${encodeURIComponent(params[key])}` : ''
    )
    .filter(Boolean)
    .join('&')}`
}
```

Reply:
Definitely do not want to remove this rule, quite the contrary - I want it to be more strict! I just do not want the `excludeArrowFunctions` to be enabled at all.